motion_tracker/data_setup/image_cropping.py

```diff
@@ -0,0 +1,210 @@
+    box_for_crop_y1 = box_for_crop_y_center + 0.5 * box_for_crop_height
+    box_for_crop = Coords(box_for_crop_x0, box_for_crop_y0, box_for_crop_x1, box_for_crop_y1)
+    return box_for_crop
+
+def get_random_crop_coords(box_coords, img_coords):
+    '''Calculate the boundaries for a new crop and the coords of the correct bounding box
+    in that new crop.
+
+    Args:
+    -----
+    box_coords: Coords object with location of bounding box in original img
+    img_coords: Coords (basically the size) of original image
+    output_img_width: Desired output width. Does not affect crop, but rather
+        the cropped area is resized to this width
+    output_img_height: Desired output height. Does not affect crop, but rather
+        the cropped area is resized to this height
+
+    Outputs:
+    --------
+    cropped_area_coords: Coords object with boundaries of the random crop
+    new_box_coords: Coords object with location of correct bounding box in
+        newly cropped image
+    '''
+
+    # box_for_crop denotes the hypothetical bounding box used to create the crop
+    # cropped_area denotes the cropped image data (including area around box_for_crop)
+    # new_box denotes the true bounding box after the crop (with coords based on crop_area)
+    box_for_crop = get_random_box_to_determine_crop(box_coords)
+
+    # calc boundary coords of cropped area. Crop_img is twice as big as box_for_crop
+    cropped_area_x0 = max(box_for_crop.x_center - box_for_crop.width, 0)
+    cropped_area_y0 = max(box_for_crop.y_center - box_for_crop.height, 0)
+    cropped_area_x1 = min(box_for_crop.x_center + box_for_crop.width, img_coords.width)
+    cropped_area_y1 = min(box_for_crop.y_center + box_for_crop.height, img_coords.height)
+    cropped_area_coords = Coords(cropped_area_x0, cropped_area_y0,
+                                 cropped_area_x1, cropped_area_y1)
+
+    box_after_crop = update_box_after_crop(box_coords, cropped_area_coords)
+    return cropped_area_coords, box_after_crop
+
+def read_bbox_data(raw_image_dir, parsed_bb_path='work/imagenet/parsed_bb.csv'):
+
+    # paper says they use .66, but they also have search area that is twice bounding
+    # box. So, 0.5 may be more appropriate.
+    max_box_frac_of_width =  0.66
+    max_box_frac_of_height = 0.66
+    images_successfully_downloaded = set(os.listdir('data/imagenet/images'))
+    bbox_df = (pd.read_csv('work/imagenet/parsed_bb.csv')
+                    .assign(box_height = lambda df: df.ymax - df.ymin,
+                            box_width = lambda df: df.xmax - df.xmin)
+                    .assign(box_frac_of_height = lambda df: df.box_height / df.height,
+                            box_frac_of_width = lambda df:  df.box_width / df.width)
+                    .query('filename in @images_successfully_downloaded')
+                    .query('box_frac_of_height < @max_box_frac_of_height')
+                    .query('box_frac_of_width < @max_box_frac_of_width'))
```

Comment:
Should this be a `>=` in the last two queries? They said that the bounding box should fill **at least** 66% of the...